obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+                if hdr == 'kevnm':
+                    # assumes users will not include a 'kevnm2' key
+                    #XXX check for empty or null value?
+                    kevnm = '{:<8s}'.format(value[0:8])
+                    kevnm2 = '{:<8s}'.format(value[8:16])
+                    hs[1] = kevnm.encode('ascii', 'strict')
+                    hs[2] = kevnm2.encode('ascii', 'strict')
+                else:
+                    hs[HD.STRHDRS.index(hdr)] = value.encode('ascii', 'strict')
+            else:
+                raise ValueError("Unrecognized header name: {}.".format(hdr))
+
+    return hf, hi, hs
+
+
+def validate_sac_content(hf, hi, hs, data, *tests):
+    """
+    Check validity of loaded SAC file content, such as header/data consistency.
+
+    Parameters
+    ----------
+    hf, hi, hs: numpy.ndarray
+        Float, int, string SAC header arrays, respectively.
+    data : numpy.ndarray or None
+        SAC data array.
+    tests : str {'delta', 'logicals', 'data_hdrs', 'enums', 'reftime', 'reltime'}
+        Perform one or more of the following validity tests:
+
+        'delta' : Time step "delta" is positive.
+        'logicals' : Logical values are 0, 1, or null
+        'data_hdrs' : Length, min, mean, max of data array match header values.
+        'enums' : Check validity of enumerated values.
+        'reftime' : Reference time values in header are all set.
+        'reltime' : Relative time values in header can be absolutely referenced.
+        'all' : Do all tests.
+
+    Raises
+    ------
+    SacInvalidContentError
+        Any of the specified tests fail.
+    ValueError
+        'data_hdrs' is specified and data is None, empty array
+        No tests specified.
+
+    """
+    # TODO: move this to util.py and write and use individual test functions,
+    # so that all validity checks are in one place?
+    ALL = ('delta', 'logicals', 'data_hdrs', 'enums', 'reftime', 'reltime')
+
+    if 'all' in tests:
+        tests = ALL
+
+    if not tests:
+        raise ValueError("No validation tests specified.")
+    elif any([(itest not in ALL) for itest in tests]):
+        msg = "Unrecognized validataion test specified"
+        raise ValueError(msg)
+
+    if 'delta' in tests:
+        val = hf[HD.FLOATHDRS.index('delta')]
+        if not (val >= 0.0):
+            msg = "Header 'delta' must be >= 0."
+            raise SacInvalidContentError(msg)
+
+    if 'logicals' in tests:
+        for hdr in ('leven', 'lpspol', 'lovrok', 'lcalda'):
+            val = hi[HD.INTHDRS.index(hdr)]
+            if val not in (0, 1, HD.INULL):
+                msg = "Header '{}' must be {{{}, {}, {}}}."
+                raise SacInvalidContentError(msg.format(hdr, 0, 1, HD.INULL))
+
+    if 'data_hdrs' in tests:
+        try:
+            isMIN = hf[HD.FLOATHDRS.index('depmin')] == data.min()
+            isMAX = hf[HD.FLOATHDRS.index('depmax')] == data.max()
+            isMEN = hf[HD.FLOATHDRS.index('depmen')] == data.mean()
```

Comment:
Only male variable names are allowed. :wink:
(Keeping parallel with the header name.)